### PR TITLE
Adding TestCases

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Improving Test coverage

## Documentation
N/A

## Certification
N/A

## Automation tests
 - Unit tests 
   coverage-86%

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java [hunk 1]
```diff
@@ -915,4 +915,269 @@ public void receive(Event[] events) {
...
+
+        SiddhiTestHelper.waitForEvents(waitTime, 2, count, timeout);
+        //assert event count
+        AssertJUnit.assertEquals("Number of events", 2, count.get());
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest15() throws InterruptedException {
+        log.info("test JsonSourceMapper with test json missing attribute");
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java:973]: Where is the assertion? If you do not get an exception try to assert log content. Else no point in having the test

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java [hunk 2]
```diff
@@ -915,4 +915,269 @@ public void receive(Event[] events) {
...
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager.createSiddhiAppRuntime(streams + query);
+        siddhiAppRuntime.start();
+        InMemoryBroker.publish("stock", "12");
+
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest16() throws InterruptedException {
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java:995]: Where is the assertion? If you do not get an exception try to assert log content. Else no point in having the test

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java [hunk 3]
```diff
@@ -915,4 +915,269 @@ public void receive(Event[] events) {
...
+                "         \"symbol\":\"WSO2\",\n" +
+                "         \"price\":55,\n" +
+                "         \"volume\":100\n" +
+                "      }\n" +
+                " }");
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest19() throws InterruptedException {
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java:1123]: assertion??

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java [hunk 4]
```diff
@@ -915,4 +915,269 @@ public void receive(Event[] events) {
...
+                "         \"price\":55.0,\n" +
+                "         \"volume\":100.0\n" +
+                "      }\n" +
+                " }");
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapper20() throws InterruptedException {
+        log.info("test JsonSourceMapper with test Test values type boolean");
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java:1154]: assertion?

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java [hunk 5]
```diff
@@ -915,4 +915,269 @@ public void receive(Event[] events) {
...
+                "         \"symbol\":\"true\",\n" +
+                "         \"price\":55,\n" +
+                "         \"volume\":100.0\n" +
+                "      }\n" +
+                " }");
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest() throws InterruptedException {
```
Comment (tishan89) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java]: assertion

**Review by tishan89 — CHANGES_REQUESTED** [review]

**Review by tishan89 — APPROVED** [review]

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java [hunk 6]
```diff
@@ -915,4 +915,291 @@ public void receive(Event[] events) {
...
+                "         \"volume\":100.0\n" +
+                "      }\n" +
+                " }");
+        AssertJUnit.assertTrue(appender.getMessages().contains("contains incompatible attribute types and values"));
+        siddhiAppRuntime.shutdown();
+    }
+
+    @Test
+    public void jsonSourceMapperTest21() throws InterruptedException {
+        log.info("test JsonSourceMapper with test json object type");
```
Comment (dimslee) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java:1184]: Why no Asserstion?

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java (hunk 4 above)
Comment (dimslee) [reply component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/JsonSourceMapperTestCase.java:1154]: +1. We need assertions.

File: component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/UnitTestAppender.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ *  under the License.
+ *
+ */
+
+package org.wso2.extension.siddhi.map.json.sourcemapper;
+
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.spi.LoggingEvent;
+
+public class UnitTestAppender extends AppenderSkeleton {
```
Comment (Megala21) [inline component/src/test/java/org/wso2/extension/siddhi/map/json/sourcemapper/UnitTestAppender.java:25]: All the public classes should be having java doc comment

**Pull request merged**